projects/igniteui-angular/src/lib/directives/focus-trap/focus-trap.directive.ts

```diff
@@ -0,0 +1,108 @@
+import { AfterViewInit, Directive, ElementRef, Input, NgModule, OnDestroy } from '@angular/core';
+import { fromEvent, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { PlatformUtil } from '../../core/utils';
+
+@Directive({
+    selector: '[igxFocusTrap]'
+})
+export class IgxFocusTrapDirective implements AfterViewInit, OnDestroy {
+    /** @hidden */
+    public get element(): HTMLElement | null {
+        return this.elementRef.nativeElement;
+    }
+
+    private destroy$ = new Subject();
+    private _focusTrap: boolean;
+
+    /** @hidden */
+    constructor(
+        private elementRef: ElementRef,
+        protected platformUtil: PlatformUtil) {
+    }
+
+    /**
+     * Sets the type of the button.
+     *
+     * @example
+     * ```html
+     * <button igxButton="icon"></button>
+     * ```
+     */
+    @Input('igxFocusTrap')
+    public set focusTrap(focusTrap: boolean) {
+        this._focusTrap = focusTrap || true;
```

Comment:
this will always set `_focusTrap` to true, whatever you pass to the setter. To set default value set the field to true like this:
```
private _focusTrap = true;
```